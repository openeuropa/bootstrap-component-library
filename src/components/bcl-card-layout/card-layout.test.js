import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoDataGroup from "@openeuropa/bcl-data-card-layout/data--group";
import demoDataGrid from "@openeuropa/bcl-data-card-layout/data--grid";
import demoDataMasonry from "@openeuropa/bcl-data-card-layout/data--masonry";

const template = "@oe-bcl/bcl-card-layout/card-layout.html.twig";
const render = (params) => renderTwigFileAsNode(template, params, true);

describe("OE - Card", () => {
  test(`group renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoDataGroup)).resolves.toMatchSnapshot();
  });
  test(`grid renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoDataGrid)).resolves.toMatchSnapshot();
  });
  test(`group renders correctly with extra col classes`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoDataGroup, cols_extra_classes: "col-class" })
    ).resolves.toMatchSnapshot();
  });
  test(`grid renders correctly with 2 per row`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoDataGrid, grid_columns: "2" })
    ).resolves.toMatchSnapshot();
  });
  test(`grid renders correctly with cards having equal height`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoDataGrid, equal_height: true })
    ).resolves.toMatchSnapshot();
  });
  test(`masonry renders correctly`, () => {
    expect.assertions(1);

    return expect(render(demoDataMasonry)).resolves.toMatchSnapshot();
  });
});
