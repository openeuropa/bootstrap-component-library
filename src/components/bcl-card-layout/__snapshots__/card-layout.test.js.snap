// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Card grid renders correctly 1`] = `
<jest>
  <div
    class="row row-cols-1 row-cols-md-3 g-2"
  >
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 1
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1002/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 1
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 1
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 1
          </p>
          <button
            class="btn btn-primary"
          >
            Custom action
          </button>
        </div>
        <div
          class="card-footer"
        >
          Footer of card 1
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 2
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1003/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 2
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 2
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 2
          </p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
        <div
          class="card-footer"
        >
          Footer of card 2
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 3
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1004/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 3
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 3
          </h6>
          <p
            class="card-text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
          <p>
            Custom text
          </p>
        </div>
        <div
          class="card-footer"
        >
          Footer of card 3
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 4
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1005/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 4
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 4
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 4
          </p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
        <div
          class="card-footer"
        >
          Footer of card 4
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 5
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1006/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 5
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 5
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 5
          </p>
        </div>
        <div
          class="card-footer"
        >
          Footer of card 5
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 6
        </div>
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 6
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 6
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 6
          </p>
          <p>
            Some representative placeholder content for the first card.
          </p>
        </div>
        <img
          alt="alt img"
          class="card-img-bottom"
          src="https://picsum.photos/seed/1007/600/400"
        />
        <div
          class="card-footer"
        >
          Footer of card 6
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Card grid renders correctly with 2 per row 1`] = `
<jest>
  <div
    class="row row-cols-2 row-cols-md-3 g-2"
  >
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 1
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1002/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 1
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 1
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 1
          </p>
          <button
            class="btn btn-primary"
          >
            Custom action
          </button>
        </div>
        <div
          class="card-footer"
        >
          Footer of card 1
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 2
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1003/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 2
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 2
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 2
          </p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
        <div
          class="card-footer"
        >
          Footer of card 2
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 3
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1004/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 3
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 3
          </h6>
          <p
            class="card-text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
          <p>
            Custom text
          </p>
        </div>
        <div
          class="card-footer"
        >
          Footer of card 3
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 4
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1005/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 4
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 4
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 4
          </p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
        <div
          class="card-footer"
        >
          Footer of card 4
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 5
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1006/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 5
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 5
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 5
          </p>
        </div>
        <div
          class="card-footer"
        >
          Footer of card 5
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-header"
        >
          Header of card 6
        </div>
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 6
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 6
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 6
          </p>
          <p>
            Some representative placeholder content for the first card.
          </p>
        </div>
        <img
          alt="alt img"
          class="card-img-bottom"
          src="https://picsum.photos/seed/1007/600/400"
        />
        <div
          class="card-footer"
        >
          Footer of card 6
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Card grid renders correctly with cards having equal height 1`] = `
<jest>
  <div
    class="row row-cols-1 row-cols-md-3 g-2"
  >
    <div
      class="col"
    >
      <div
        class="mb-3 card h-100"
      >
        <div
          class="card-header"
        >
          Header of card 1
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1002/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 1
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 1
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 1
          </p>
          <button
            class="btn btn-primary"
          >
            Custom action
          </button>
        </div>
        <div
          class="card-footer"
        >
          Footer of card 1
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card h-100"
      >
        <div
          class="card-header"
        >
          Header of card 2
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1003/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 2
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 2
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 2
          </p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
        <div
          class="card-footer"
        >
          Footer of card 2
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card h-100"
      >
        <div
          class="card-header"
        >
          Header of card 3
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1004/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 3
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 3
          </h6>
          <p
            class="card-text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
          <p>
            Custom text
          </p>
        </div>
        <div
          class="card-footer"
        >
          Footer of card 3
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card h-100"
      >
        <div
          class="card-header"
        >
          Header of card 4
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1005/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 4
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 4
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 4
          </p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
        <div
          class="card-footer"
        >
          Footer of card 4
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card h-100"
      >
        <div
          class="card-header"
        >
          Header of card 5
        </div>
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1006/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 5
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 5
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 5
          </p>
        </div>
        <div
          class="card-footer"
        >
          Footer of card 5
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card h-100"
      >
        <div
          class="card-header"
        >
          Header of card 6
        </div>
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 6
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 6
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 6
          </p>
          <p>
            Some representative placeholder content for the first card.
          </p>
        </div>
        <img
          alt="alt img"
          class="card-img-bottom"
          src="https://picsum.photos/seed/1007/600/400"
        />
        <div
          class="card-footer"
        >
          Footer of card 6
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Card group renders correctly 1`] = `
<jest>
  <div
    class="card-group"
  >
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        Header of card 1
      </div>
      <img
        alt="alt img"
        class="card-img-top"
        src="https://picsum.photos/seed/1008/600/400"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Title card 1
        </h5>
        <h6
          class="card-subtitle mb-2"
        >
          Subtitle card 1
        </h6>
        <p
          class="card-text mb-2"
        >
          Text card with longer text which is supported 1
        </p>
        <button
          class="btn btn-primary"
        >
          Custom action
        </button>
      </div>
      <div
        class="card-footer"
      >
        Footer of card 1
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        Header of card 2
      </div>
      <img
        alt="alt img"
        class="card-img-top"
        src="https://picsum.photos/seed/1009/600/400"
      />
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Title card 2
        </h5>
        <h6
          class="card-subtitle mb-2"
        >
          Subtitle card 2
        </h6>
        <p
          class="card-text mb-2"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
        </p>
        <p>
          Custom text
        </p>
      </div>
      <div
        class="card-footer"
      >
        Footer of card 2
      </div>
    </div>
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        Header of card 3
      </div>
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Title card 3
        </h5>
        <h6
          class="card-subtitle mb-2"
        >
          Subtitle card 3
        </h6>
        <p
          class="card-text mb-2"
        >
          Text card with longer text which is supported 3
        </p>
        <p>
          Some representative placeholder content for the first card.
        </p>
      </div>
      <img
        alt="alt img"
        class="card-img-bottom"
        src="https://picsum.photos/seed/1010/600/400"
      />
      <div
        class="card-footer"
      >
        Footer of card 3
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Card masonry renders correctly 1`] = `
<jest>
  <div
    class="row row-cols-1 row-cols-md-3"
    data-masonry="{\\"percentPosition\\": true }"
  >
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 1
          </h5>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 1
          </p>
          <button
            class="btn btn-primary"
          >
            Custom action
          </button>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 2
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 2
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 2
          </p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
        </div>
        <img
          alt="alt img"
          class="card-img-bottom"
          src="https://picsum.photos/seed/1011/600/400"
        />
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 3
          </h5>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 3
          </p>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 3
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 3
          </h6>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 3
          </p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <img
          alt="alt img"
          class="card-img-top"
          src="https://picsum.photos/seed/1012/600/400"
        />
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 4
          </h5>
          <h6
            class="card-subtitle mb-2"
          >
            Subtitle card 4
          </h6>
          <p
            class="card-text mb-2"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
          <button
            class="btn btn-success"
          >
            Custom action
          </button>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 text-center card"
      >
        <img
          alt="alt img"
          class="card-img"
          src="https://picsum.photos/seed/1008/600/400"
        />
        <div
          class="card-img-overlay"
        >
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Title card 5
            </h5>
            <h6
              class="card-subtitle mb-2"
            >
              Subtitle card 5
            </h6>
            <p
              class="card-text mb-2"
            >
              Text card with longer text which is supported 5
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col"
    >
      <div
        class="mb-3 card"
      >
        <div
          class="card-body"
        >
          <h5
            class="card-title"
          >
            Title card 6
          </h5>
          <p
            class="card-text mb-2"
          >
            Text card with longer text which is supported 6
          </p>
          <p>
            Some representative placeholder content for the first card.
          </p>
        </div>
      </div>
    </div>
  </div>
</jest>
`;
